tar/bsdtar.c

```diff
@@ -219,6 +221,8 @@ main(int argc, char **argv)
 	bsdtar->filename = getenv("TAPE");
 	if (bsdtar->filename == NULL)
 		bsdtar->filename = _PATH_DEFTAPE;
+    if (!access(bsdtar->filename, F_OK))
```

Comment:
A slightly cleaner way to organize this might be to:
1. Rework the definition for `_PATH_DEFTAPE` above into a standard `#if`...`#elif`...`#else` chain.
2. For systems that lack a default tape drive (Apple), this will let you leave `_PATH_DEFTAPE` undefined,
3. You can then handle the default tape drive as follows
```
   bsdtar->filename = getenv("TAPE");
#if defined(_PATH_DEFTAPE)
   if (bsdtar->filename == NULL && access(_PATH_DEFTAPE, F_OK)) {
      bsdtar->filename = _PATH_DEFTAPE;
   }
#endif
  if (bsdtar->filename == NULL) {
     bsdtar->filename = _PATH_FALLBACK;
  }
```

Reply:
I would also suggest renaming `_PATH_FALLBACK` to `_PATH_STDOUT`.  With the logic just above, we would then have:
* Use the value from the `TAPE` environment variable if that exists
* If `_PATH_DEFTAPE` was defined at compile time and the corresponding path exists, use that.  In essence, this just makes `_PATH_DEFTAPE` into a build-time option to set a default value for the `TAPE` environment variable.
* Otherwise, default to stdout

With the above logic, we can then discuss changing `_PATH_DEFTAPE` to be undefined on some or all systems.  Regardless, we can leave the above logic in place for people who want to explicitly set `_PATH_DEFTAPE` in their build environment.

Reply:
Thanks @kientzle for your feedback. I like your suggestion. The only thing I would change is `_PATH_STDOUT`, since "-" will be used for both `stdin` and `stdout`. What about  `_PATH_STDIN_STDOUT`?

Reply:
That sounds like a good way to handle that.   Or `_PATH_STDIO` might also work.

Reply:
Thanks again for your feedback. I updated the PR based on your comment. 

The only issue I see is that if TAPE env is set it will be use without checking it the path is valid or not. Most probably systems that don't use tape should not set that variable, so it should be fine.